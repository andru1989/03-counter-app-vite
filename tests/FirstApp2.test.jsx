import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Tests in <FirstApp />', () => {

  const title = 'Hi, my name is Andres';
  const subTitle = 'Subtitle test';

  test('should match snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test('should show the title Hi, my name is Andres', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('should show the title in h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title);
  });

  test('should show the subtitle sent by props', () => {
    render(
      <FirstApp
        title={title}
        subTitle={subTitle}
      />
    );

    expect(screen.getByText(subTitle)).toBeTruthy();
  });

});