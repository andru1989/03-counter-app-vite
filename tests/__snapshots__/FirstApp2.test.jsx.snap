// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hi, my name is Andres
  </h1>
  <p>
    No hay subt√≠tulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
